x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/activity_log_date_range_picker/index.tsx

```diff
@@ -29,95 +41,102 @@ const StickyFlexItem = styled(EuiFlexItem)`
 
 export const DateRangePicker = memo(() => {
   const dispatch = useDispatch();
-  const { page, pageSize, startDate, endDate, isInvalidDateRange } = useEndpointSelector(
+  const { page, pageSize, startDate, endDate, autoRefreshOptions } = useEndpointSelector(
     getActivityLogDataPaging
   );
 
-  const onClear = useCallback(
-    ({ clearStart = false, clearEnd = false }: { clearStart?: boolean; clearEnd?: boolean }) => {
-      dispatch({
-        type: 'endpointDetailsActivityLogUpdatePaging',
-        payload: {
-          disabled: false,
-          page,
-          pageSize,
-          startDate: clearStart ? undefined : startDate,
-          endDate: clearEnd ? undefined : endDate,
-        },
-      });
-    },
-    [dispatch, endDate, startDate, page, pageSize]
-  );
+  const [recentlyUsedRanges, setRecentlyUsedRanges] = useState<EuiSuperDatePickerRecentRange[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const stopLoading = useCallback(() => {
+    setIsLoading(false);
+  }, [setIsLoading]);
+
+  const startLoading = useCallback(() => {
+    setTimeout(stopLoading, 1000);
```

Comment:
Could we avoid this `setTimeout()` and instead tie it to the API response having been received (which I think is what the purpose of it is, right)?

A concern I have here is that this will always  call `stopLoading()` even if the component has been unmounted, which will cause some errors in the console. But after looking closer at how its being used, I think both of these callbacks could be collapsed into just a single one - as:

```typescript
const startLoading = useCallback(async () => {
  setIsLoading(true);
  await new Promise(r => setTimeout(r, 1000));
  setIsLoading(false);
}, [])
```

You could even move the `dispatch()` on line 100 here (change the callback to accept start/end dates), thus this callback would actually initiate the loading.

Reply:
done https://github.com/elastic/kibana/pull/108722/commits/b638c2df443657be3983ee9780af319235bde2ea#diff-76b29049fb481f2017c39144bb7f7ea6df0c5f927d64e2c6a2d6bda677cc1798R55